models/schedule/factory.py

```diff
@@ -0,0 +1,19 @@
+from datetime import datetime, timedelta
+from models.schedule.kinds.cron import Cron
+from models.schedule.kinds.interval import Interval
+
+
+class Factory:
+    def __init__(self, json_schedule):
+        if json_schedule["kind"] == "interval":
+            self.kind = Interval(
+                timedelta(minutes=json_schedule["time"]),
+                datetime.strptime(json_schedule["start"], "%H:%M").time(),
+                datetime.strptime(json_schedule["end"], "%H:%M").time(),
+            )
+        elif json_schedule["kind"] == "cron":
+            self.kind = Cron(json_schedule["cron"])
```

Comment:
```suggestion
class Factory:
    def __init__(self, json_schedule):
        self.__json_schedule = json_schedule
        
    def new(self):
        kind = self.__json_schedule["kind"]
        
        if kind == "interval":
            return Interval(self.__json_schedule)
        elif kind == "cron":
            return Cron(self.__json_schedule)
```

Algo nessa ideia, talvez até adicionar um else para levantar uma exception quando não for um desses

Reply:
Considerado no ajuste do factory